marimo/_islands/island_generator.py

```diff
@@ -51,26 +66,53 @@ def code(self) -> str:
 
     def render(
         self,
-        include_code: bool = True,
-        include_output: bool = True,
+        display_code: Optional[bool] = None,
+        display_output: Optional[bool] = None,
+        is_reactive: Optional[bool] = None,
     ) -> str:
         """
         Render the HTML island code for the cell.
+        Note: This will override construction defaults.
 
         *Args:*
 
-        - include_code (bool): Whether to include the code in the HTML.
-        - include_output (bool): Whether to include the output in the HTML.
+        - display_code (bool): Whether to display the code in HTML.
+        - display_output (bool): Whether to include the output in the HTML.
+        - is_reactive (bool): Whether this code block will run with pyodide.
 
         *Returns:*
 
         - str: The HTML code.
         """
-        if include_code is False and include_output is False:
+        is_reactive = (
+            is_reactive if is_reactive is not None else self._is_reactive
+        )
+        display_code = (
+            display_code if display_code is not None else self._display_code
+        )
+        display_output = (
+            display_output
+            if display_output is not None
+            else self._display_output
+        )
+
+        if not (display_code or display_output or is_reactive):
             raise ValueError("You must include either code or output")
 
         output = handle_mimetypes(self.output) if self.output else None
 
+        # Specifying display_code=False will hide the code block, but still
+        # make it present for reactivity, unless reactivity is disabled.
+        if display_code:
+            # TODO: Allow for non-disabled code editors.
+            code_block = as_html(code_editor(self.code, disabled=False)).text
```

Comment:
So the code as is, just needs to flip `disabled=False` to `disabled=True` and the editor works.

I left it as is, because I think it needs some UI love (maybe the stale indicator and an info icon?)

Reply:
woah that is pretty impressive! yea i can take a design pass